packages/angular_devkit/core/src/jobs/simple-registry_spec.ts

```diff
@@ -0,0 +1,578 @@
+
+    expect(done.length).toBe(0);
+
+    await job1.output.toPromise();
+    expect(done).toContain(1);
+    expect(done).not.toContain(4);
+    expect(done).not.toContain(7);
+
+    await job5.output.toPromise();
+    expect(done).toContain(1);
+    expect(done).toContain(2);
+    expect(done).not.toContain(4);
+    expect(done).toContain(5);
+    expect(done).not.toContain(7);
+
+    await job7.output.toPromise();
+    expect(done.length).toBe(7);
+    // Might be out of order.
+    expect(done).toEqual(jasmine.arrayContaining([1, 2, 3, 4, 5, 6, 7]));
+    // Verify at least partial order.
+    expect(done[done.length - 1]).toBe(7);
+    expect(done.indexOf(4)).toBeGreaterThan(done.indexOf(1));
+    expect(done.indexOf(5)).toBeGreaterThan(done.indexOf(2));
+    expect(done.indexOf(6)).toBeGreaterThan(done.indexOf(3));
+  });
+
+  it('does not start dependencies until the last one is subscribed to', async () => {
+    // This test creates the following graph of dependencies:
+    //     1  <-.--  2  <-.--  4  <-.---------.--  6
+    //                    +--  3  <-+--  5  <-'
+    // Which can result only in the execution orders: [1, 2, 3, 4, 5, 6]
+    // Only subscribe to the last one.
+
+    const started: number[] = [];
+    const done: number[] = [];
+
+    registry.register(
+      'job',
+      createJobHandler<number, number, number>((argument: number) => {
+        started.push(argument);
+
+        return new Promise(resolve => setImmediate(() => {
+          done.push(argument);
+          resolve(argument);
+        }));
+      }),
+      { argument: true, output: true },
+    );
+
+    // Run jobs.
+    const job1 = scheduler.schedule('job', 1);
+    const job2 = scheduler.schedule('job', 2, { dependencies: job1 });
+    const job3 = scheduler.schedule('job', 3, { dependencies: job2 });
+    const job4 = scheduler.schedule('job', 4, { dependencies: [job2, job3] });
+    const job5 = scheduler.schedule('job', 5, { dependencies: [job1, job2, job4] });
+    const job6 = scheduler.schedule('job', 6, { dependencies: [job4, job5] });
+
+    // Just subscribe to the last job in the lot.
+    job6.rawOutput.subscribe();
+    // Expect the first one to start.
+    expect(started).toEqual([1]);
+    // Wait for the first one to finish.
+    await job1.output.toPromise();
+    // Expect the second one to have started, and the first one to be done.
+    expect(started).toEqual([1, 2]);
+    expect(done).toEqual([1]);
+
+    // Rinse and repeat.
+    await job2.output.toPromise();
+    expect(started).toEqual([1, 2, 3]);
+    expect(done).toEqual([1, 2]);
+
+    await job3.output.toPromise();
+    expect(started).toEqual([1, 2, 3, 4]);
+    expect(done).toEqual([1, 2, 3]);
+
+    await job4.output.toPromise();
+    expect(started).toEqual([1, 2, 3, 4, 5]);
+    expect(done).toEqual([1, 2, 3, 4]);
+
+    // Just skip job 5.
+    await job6.output.toPromise();
+    expect(done).toEqual(started);
+  });
+
+  it('can be paused', async () => {
```

Comment:
It's worth specifying in tests what will happen if multiple jobs pause the scheduler. Do all of them have to resume it after that or only one resume is enough?

Reply:
There's another test for that behaviour actually, the `'can be paused (multiple)'` one.

It's also mentioned in the readme:
```
   * Pause the scheduler, temporary queueing _new_ jobs. Returns a resume function that should be
   * used to resume execution. If multiple `pause()` were called, all their resume functions must
   * be called before the Scheduler actually starts new jobs. Additional calls to the same resume
   * function will have no effect.
```

Reply:
Yes. We test for this and the behavior is documented. Do you feel there's a missing documentation somewhere?